planning/behavior_velocity_planner/include/scene_module/dynamic_obstacle_stop/scene.hpp

```diff
@@ -0,0 +1,170 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SCENE_MODULE__DYNAMIC_OBSTACLE_STOP__SCENE_HPP_
+#define SCENE_MODULE__DYNAMIC_OBSTACLE_STOP__SCENE_HPP_
+
+#include "scene_module/dynamic_obstacle_stop/debug.hpp"
+#include "scene_module/dynamic_obstacle_stop/dynamic_obstacle.hpp"
+#include "scene_module/dynamic_obstacle_stop/utils.hpp"
+#include "scene_module/scene_module_interface.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace behavior_velocity_planner
+{
+namespace bg = boost::geometry;
+using autoware_auto_perception_msgs::msg::PredictedObjects;
+using autoware_auto_planning_msgs::msg::PathPointWithLaneId;
+using autoware_auto_planning_msgs::msg::PathWithLaneId;
+using dynamic_obstacle_stop_utils::DetectionAreaSize;
+using dynamic_obstacle_stop_utils::PlannerParam;
+using dynamic_obstacle_stop_utils::State;
+using tier4_debug_msgs::msg::Float32Stamped;
+using BasicPolygons2d = std::vector<lanelet::BasicPolygon2d>;
+
+class DynamicObstacleStopModule : public SceneModuleInterface
+{
+public:
+  DynamicObstacleStopModule(
+    const int64_t module_id, const std::shared_ptr<const PlannerData> & planner_data,
+    const PlannerParam & planner_param, const rclcpp::Logger logger,
+    std::unique_ptr<DynamicObstacleCreator> dynamic_obstacle_creator,
+    const std::shared_ptr<DynamicObstacleStopDebug> & debug_ptr,
+    const rclcpp::Clock::SharedPtr clock);
+
+  bool modifyPathVelocity(
+    autoware_auto_planning_msgs::msg::PathWithLaneId * path,
+    tier4_planning_msgs::msg::StopReason * stop_reason) override;
+
+  visualization_msgs::msg::MarkerArray createDebugMarkerArray() override;
+
+  void setPlannerParam(const PlannerParam & planner_param);
+
+private:
+  // Parameter
+  PlannerParam planner_param_;
+
+  // Variable
+  State state_{State::GO};
+  rclcpp::Time stop_time_;
+  BasicPolygons2d partition_lanelets_;
+  std::unique_ptr<DynamicObstacleCreator> dynamic_obstacle_creator_;
+  std::shared_ptr<DynamicObstacleStopDebug> debug_ptr_;
+
+  // Function
+  boost::optional<std::vector<DynamicObstacle>> createDynamicObstacles(
+    const PredictedObjects & predicted_objects,
+    const pcl::PointCloud<pcl::PointXYZ>::ConstPtr & points, const PathWithLaneId & path,
+    const geometry_msgs::msg::Pose & current_pose, const std::string & detection_method) const;
+
+  std::vector<DynamicObstacle> createDynamicObstaclesFromObjects(
+    const PredictedObjects & predicted_objects) const;
+
+  std::vector<DynamicObstacle> createDynamicObstaclesFromPoints(
+    const pcl::PointCloud<pcl::PointXYZ> & obstacle_pointcloud, const PathWithLaneId & path) const;
+
+  std::vector<DynamicObstacle> createDynamicObstaclesFromObjects(
+    const PredictedObjects & predicted_objects, const PathWithLaneId & path) const;
```

Comment:
Can these lines be deleted?

Reply:
Yes. deleted in https://github.com/autowarefoundation/autoware.universe/pull/752/commits/0cb971986888c695a4add54e8b9144d497da2707.